src/hipscat/pixel_tree/pixel_alignment.py

```diff
@@ -0,0 +1,176 @@
+            outer - use all pixels from both catalogs
+    Returns:
+        The `PixelAlignment` object with the alignment from the two trees
+    """
+
+    tree_builder = PixelTreeBuilder()
+    pixels_to_search = _get_children_pixels_from_trees([left, right],
+                                                            left.root_pixel.pixel)
+
+    while len(pixels_to_search) > 0:
+        search_pixel = pixels_to_search.pop(0)
+        if search_pixel in left and search_pixel in right:
+            left_node = left[search_pixel]
+            right_node = right[search_pixel]
+            if left_node.node_type == right_node.node_type:
+                if left_node.node_type == PixelNodeType.LEAF:
+                    tree_builder.create_node_and_parent_if_not_exist(search_pixel,
+                                                                     PixelNodeType.LEAF)
+                else:
+                    pixels_to_search += _get_children_pixels_from_trees([left, right], search_pixel)
+            else:
+                if left_node.node_type == PixelNodeType.LEAF:
+                    tree_with_leaf_node = "left"
+                    inner_node = right_node
+                else:
+                    tree_with_leaf_node = "right"
+                    inner_node = left_node
+                if _should_include_all_pixels_from_tree(tree_with_leaf_node, alignment_type):
+                    tree_builder.create_node_and_parent_if_not_exist(search_pixel,
+                                                                     PixelNodeType.LEAF)
+                    tree_builder.split_leaf_to_match_partitioning(inner_node)
+                else:
+                    tree_builder.create_node_and_parent_if_not_exist(search_pixel,
+                                                                     PixelNodeType.INNER)
+                    tree_builder.add_all_descendants_from_node(inner_node)
+        elif search_pixel in left and search_pixel not in right:
+            if _should_include_all_pixels_from_tree("left", alignment_type):
+                tree_builder.create_node_and_parent_if_not_exist(search_pixel,
+                                                                 left[search_pixel].node_type)
+                tree_builder.add_all_descendants_from_node(left[search_pixel])
+        elif search_pixel in right and search_pixel not in left:
+            if _should_include_all_pixels_from_tree("right", alignment_type):
+                tree_builder.create_node_and_parent_if_not_exist(search_pixel,
+                                                                 right[search_pixel].node_type)
+                tree_builder.add_all_descendants_from_node(right[search_pixel])
+    tree = tree_builder.build()
+    pixel_mapping = _generate_pixel_mapping_from_tree(left, right, tree)
+    return PixelAlignment(tree, pixel_mapping, alignment_type)
+
+
+def _get_children_pixels_from_trees(trees: List[PixelTree], pixel: HealpixInputTypes) -> \
```

Comment:
I'll add the doc strings to the functions.

In this case, you're right except we're looking for the children's HEALPix pixels and not the children directly. The function also uses a set to avoid duplicates, so it would produce a list of 4 children (If the inner node had a full set of child nodes)